2021-12-26-purenix.md

```diff
@@ -0,0 +1,61 @@
+------------------------------------------------------
+title: PureNix
+summary: Transpiling PureScript to Nix
+tags: haskell, nixos, purescript
+draft: false
+------------------------------------------------------
+
+My friend [Jonas Carpay](https://jonascarpay.com/)
+([@jonascarpay](https://github.com/jonascarpay)) and I recently released
+[PureNix](https://github.com/purenix-org/purenix).  PureNix is a Nix backend for
+[PureScript](https://www.purescript.org/).  PureNix allows you to write
+PureScript code and transpile it to Nix code.  This can be helpful when trying
+to write large or tricky code in Nix.
+
+This blog post series is all about PureNix.  It explains how to get started with
+PureNix, why we wrote PureNix, when you might want to use PureNix, and more.
+If you just want to play around with PureNix, all you need is the first post.
+
+
+1.  [Getting Started with PureNix](./2021-12-27-getting-started-with-purenix)
+
+    This post gives a high-level overview of PureNix, and explains how to setup
+    an environment for playing around with it.
+
+2.  Why PureNix? (coming soon)
+
+    This post explains why we decided to write PureNix.
+
+3.  Under the Hood of PureNix (coming soon)
+
+    This post explains how PureNix works internally.  It talks about what goes
+    in to writing a new PureScript backend.
+
+4.  PureNix vs. Nix (coming soon)
+
+    This post talks about when you should choose PureNix over writing in raw
+    Nix.
+
```

Comment:
I was thinking that 2 (Why PureNix?) would basically just be the story of how we started writing PureNix:

- I wanted to write `cabal2nixWithoutIFD`
- I thought writing PureNix would be a funny way to do it
- I talked to you about it, and you had a couple different suggestions (like a Haskell backend based on core, or just a Haskell DSL).
- After looking around a bit, doing it with PureScript seemed like it would be the easiest.
- We got PureNix done in about a month.  It worked better than either of us expected.
- We got a couple libraries ported a couple months after that.  It was a little time-consuming, but it mostly worked out well.
- Writing `cabal2nixWithoutIFD` was relatively easy.
- Basically everything worked better than we had expected.

I thought of calling this post "The History of PureNix", but it feels weird to call it a "history" when it has only existed for a few months.  Although I agree that "Why PureNix" is confusing as well.

For 4 (PureNix vs. Nix), I was thinking of showing some expressions written in both Nix and PureNix.  I was planning on trying to make the argument that for simple derivations (like when you're using Nix effectively as a bash templating language), raw Nix is easier.  But when you're writing actual code, PureNix can be easier.